### PR TITLE
input+lnwallet: update taproot scripts to accept optional aux leaf

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we update all the taproot scripts to also accept an optional aux leaf. This aux leaf can be used to add more redemption paths for advanced channels, or just as an extra commitment space. All the `input/taproot_test.go` have also been updated to ensure that the new slightly larger control blocks are always properly generated.

### CODE REVIEW

File: input/script_utils.go [hunk 1]
```diff
@@ -638,6 +639,12 @@ type HtlcScriptTree struct {
 	// TimeoutTapLeaf is the tapleaf for the timeout path.
 	TimeoutTapLeaf txscript.TapLeaf
 
+	// AuxLeaf is an axuillaruy leaf that can be used to extend the base
+	// HTLC script tree with new spend paths, or just as extra commitment
+	// space. When present, this leaf will always be in the left-most or
+	// right-most area of the tapscript tree.
+	AuxLeaf fn.Option[txscript.TapLeaf]
```
Comment (GeorgeTsagk) [inline input/script_utils.go:646]: in order to be open-ended with regards to flexibility, couldn't we have multiple aux leaves here? something like `fn.Slice`

we could hide everything under a single `AuxLeaf` but I think just having multiple leaves will keep it simpler overall on the script level itself

File: input/script_utils.go [hunk 2]
```diff
@@ -731,10 +739,15 @@ func senderHtlcTapScriptTree(senderHtlcKey, receiverHtlcKey,
 		return nil, err
 	}
 
+	tapLeaves := []txscript.TapLeaf{successTapLeaf, timeoutTapLeaf}
+	auxLeaf.WhenSome(func(l txscript.TapLeaf) {
```
Comment (guggero) [inline input/script_utils.go:743]: Is it worth noting here that if there is an odd number of leaves, then the last one will always end up being a level higher than the other leaves. Which in our case works out to exactly what we want, with just three leaves. But the assumptions here (with using `AssembleTaprootScriptTree`) would break down for let's say 5 leaves.
Then again, we never have more than 3 leaves in any LN use cases, so maybe not worth spending more time on.

File: input/script_utils.go [hunk 3]
```diff
@@ -714,7 +721,8 @@ var _ TapscriptDescriptor = (*HtlcScriptTree)(nil)
 // the HTLC key for HTLCs on the sender's commitment.
 func senderHtlcTapScriptTree(senderHtlcKey, receiverHtlcKey,
 	revokeKey *btcec.PublicKey, payHash []byte,
-	hType htlcType) (*HtlcScriptTree, error) {
+	hType htlcType,
```
Comment (guggero) [inline input/script_utils.go:724]: micro-nit: looks like `hType` could fit on previous line.

File: input/script_utils.go [hunk 4]
```diff
@@ -2060,6 +2096,14 @@ type CommitScriptTree struct {
 	// RevocationLeaf is the leaf used to spend the output with the
 	// revocation key signature.
 	RevocationLeaf txscript.TapLeaf
+
+	// AuxLeaf is an axuillaruy leaf that can be used to extend the base
+	// commitment script tree with new spend paths, or just as extra
+	// commitment space. When present, this leaf will always be in the
+	// left-most or right-most area of the tapscript tree.
+	//
+	// TODO(roasbeeF): need to ensure commitment position...
```
Comment (guggero) [inline input/script_utils.go:2105]: Is this TODO related to my comment above? About relying on the logic in `txscript.AssembleTaprootScriptTree`?

File: watchtower/blob/justice_kit_test.go [hunk 5]
```diff
@@ -460,7 +463,7 @@ func TestJusticeKitToLocalWitnessConstruction(t *testing.T) {
 				rev *btcec.PublicKey) []byte {
 
 				script, _ := input.NewLocalCommitScriptTree(
-					csvDelay, delay, rev,
+					csvDelay, delay, rev, fn.None[txscript.TapLeaf](),
```
Comment (guggero) [inline watchtower/blob/justice_kit_test.go:466]: We need similar changes in other watchtower related code, see linter failures.

**Review by guggero — COMMENTED** [review]
Very nice, super small diff for what we get out of it. It's obvious a lot of thought already went into this while refactoring for the Simple Taproot Channels :100:

File: input/script_utils.go (hunk 4 above)
Comment (Roasbeef) [reply input/script_utils.go:2105]: Yep, but as it it works since here we have it be that odd element out, which'll give it the top right position in the tree.

**Comment by guggero** [comment]
The single commit of this PR is included in https://github.com/lightningnetwork/lnd/pull/8684.
